### PR TITLE
Dmp 2158.retention v7

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-2158

https://tools.hmcts.net/jira/browse/DMP-2116

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/common/repository/CaseRetentionRepository.java [hunk 1]
```diff
@@ -19,10 +19,10 @@ public interface CaseRetentionRepository extends JpaRepository<CaseRetentionEnti
 
     @Query("""
         SELECT c
-        FROM CaseRetentionEntity c
-        WHERE courtCase = :courtCase
-        AND manualOverride=false
-        AND currentState='COMPLETE'
+        FROM CaseRetentionEntity c, RetentionPolicyTypeEntity rpt
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/repository/CaseRetentionRepository.java:22]: If you're navigating across the entities on line 24, then why would you need to add rpt here?

File: src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+
 @Repository
 public interface RetentionPolicyTypeRepository extends JpaRepository<RetentionPolicyTypeEntity, Integer> {
 
+    @Query("""
+        SELECT r
+        FROM RetentionPolicyTypeEntity r
+        WHERE fixedPolicyKey = :fixedPolicyKey
+        and policyStart <= :currentTime
+        and (policyEnd is null or policyEnd > current_time)
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java:19]: Should this be `:currentTime`?

Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java:19]: I don't think `policy_end_ts` is nullable?

Comment (richardbull) [reply src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java:19]: My mistake - looks like the ER pdf is wrong

File: src/main/java/uk/gov/hmcts/darts/common/entity/RetentionPolicyTypeEntity.java [hunk 3]
```diff
@@ -28,11 +28,14 @@ public class RetentionPolicyTypeEntity extends CreatedModifiedBaseEntity {
     private Integer id;
 
     @Column(name = "fixed_policy_key")
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/RetentionPolicyTypeEntity.java:30]: We should probably add the nullable = false to the NOT NULL columns to make it a bit easier to understand looking at our entity?

File: src/main/java/uk/gov/hmcts/darts/common/entity/CaseManagementRetentionEntity.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public class CaseManagementRetentionEntity {
     @JoinColumn(name = "eve_id", nullable = false)
     private EventEntity eventEntity;
 
-    @Column(name = "total_sentence", nullable = false)
+    @Column(name = "total_sentence")
     private String totalSentence;
 
     @Column(name = "event_ts", nullable = false)
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/CaseManagementRetentionEntity.java:47]: This column doesn't seem to match the current table definition that we seen earlier in the knowledge transfer - it is NULLABLE

File: src/main/java/uk/gov/hmcts/darts/common/repository/CaseRetentionRepository.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/repository/CaseRetentionRepository.java]: removed

File: src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/repository/RetentionPolicyTypeRepository.java]: done

File: src/main/java/uk/gov/hmcts/darts/common/entity/RetentionPolicyTypeEntity.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/RetentionPolicyTypeEntity.java]: not sure it matters, but have added in anyway

File: src/main/java/uk/gov/hmcts/darts/common/entity/CaseManagementRetentionEntity.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/CaseManagementRetentionEntity.java]: removed

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**